@adityaketkar 

I ran the tests on my system. The integration test fails with the following exception.
```

org.elasticsearch.ElasticsearchIllegalStateException: Failed to created node environment
	at org.elasticsearch.node.internal.InternalNode.<init>(InternalNode.java:162)
	at org.elasticsearch.node.NodeBuilder.build(NodeBuilder.java:159)
	at org.elasticsearch.node.NodeBuilder.node(NodeBuilder.java:166)
	at in.ac.bits.protocolanalyzer.persistence.repository.ElasticSearchConfig.elasticsearchTemplate(ElasticSearchConfig.java:32)
	at unit.in.ac.bits.protocolanalyzer.persistence.repository.ElasticSearchConfigTest.testElasticsearchTemplate(ElasticSearchConfigTest.java:47)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.springframework.test.context.junit4.statements.RunBeforeTestMethodCallbacks.evaluate(RunBeforeTestMethodCallbacks.java:74)
	at org.springframework.test.context.junit4.statements.RunAfterTestMethodCallbacks.evaluate(RunAfterTestMethodCallbacks.java:83)
	at org.springframework.test.context.junit4.statements.SpringRepeat.evaluate(SpringRepeat.java:72)
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.runChild(SpringJUnit4ClassRunner.java:233)
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.runChild(SpringJUnit4ClassRunner.java:87)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.springframework.test.context.junit4.statements.RunBeforeTestClassCallbacks.evaluate(RunBeforeTestClassCallbacks.java:61)
	at org.springframework.test.context.junit4.statements.RunAfterTestClassCallbacks.evaluate(RunAfterTestClassCallbacks.java:71)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.run(SpringJUnit4ClassRunner.java:176)
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:86)
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:539)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:761)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:461)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:207)
Caused by: java.nio.file.AccessDeniedException: /opt/darshini-es
	at sun.nio.fs.UnixException.translateToIOException(UnixException.java:84)
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102)
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:107)
	at sun.nio.fs.UnixFileSystemProvider.createDirectory(UnixFileSystemProvider.java:384)
	at java.nio.file.Files.createDirectory(Files.java:674)
	at java.nio.file.Files.createAndCheckIsDirectory(Files.java:781)
	at java.nio.file.Files.createDirectories(Files.java:767)
	at org.elasticsearch.env.NodeEnvironment.<init>(NodeEnvironment.java:105)
	at org.elasticsearch.node.internal.InternalNode.<init>(InternalNode.java:160)
	... 33 more
```

The exception clearly points to not having `/opt/darshini-es` directory in the file system. I created the same and given ownership permissions over the folder to the user who runs Eclipse. The exception disappears and all the tests pass. The integration test ends up creating data repositories for Elastic Search. We thought we were mocking the ElasticSearchTemplate, there by mocking the Elastic Search completely. Obviously, that conclusion is not true.

We might as well not mock anything in the integration tests.

--- a/src/test/java/integration/config/in/ac/bits/protocolanalyzer/mvc/model/ExperimentTestConfig.java
+++ b/src/test/java/integration/config/in/ac/bits/protocolanalyzer/mvc/model/ExperimentTestConfig.java
@@ -16,23 +16,37 @@ import in.ac.bits.protocolanalyzer.analyzer.transport.TransportAnalyzer;
 import in.ac.bits.protocolanalyzer.analyzer.transport.UdpAnalyzer;
 import in.ac.bits.protocolanalyzer.mvc.model.Experiment;
 import in.ac.bits.protocolanalyzer.persistence.repository.AnalysisRepository;
+import in.ac.bits.protocolanalyzer.persistence.repository.ElasticSearchConfig;
+import in.ac.bits.protocolanalyzer.persistence.repository.ElasticSearchFactoryImpl;
 import in.ac.bits.protocolanalyzer.persistence.repository.SaveRepository;
 import in.ac.bits.protocolanalyzer.protocol.Protocol;
 import in.ac.bits.protocolanalyzer.protocol.ProtocolChecker;
 import in.ac.bits.protocolanalyzer.protocol.ProtocolGraph;
 import in.ac.bits.protocolanalyzer.protocol.ProtocolGraphParser;
 
-import org.mockito.Mock;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.core.env.Environment;
 import org.springframework.data.elasticsearch.core.ElasticsearchTemplate;
 
 @Configuration
+@PropertySource("classpath:META-INF/elasticsearch.properties")
 public class ExperimentTestConfig {
 	
+	@Autowired 
+	Environment env;
 	
-	@Mock
-	public ElasticsearchTemplate template;
+	@Bean
+	public ElasticSearchFactoryImpl getElasticSearchFactoryImpl() {
+		return new ElasticSearchFactoryImpl();
+	}
+
+	@Bean
+	public ElasticSearchConfig getElasticSearchConfig() {
+		return new ElasticSearchConfig();
+	}
 	
 	@Bean
 	public Experiment getSampleExperiment() {
@@ -96,7 +110,7 @@ public class ExperimentTestConfig {
 	
 	@Bean
 	public ElasticsearchTemplate getSampleElasticsearchTemplate() {
-		return template;
+		return (ElasticsearchTemplate) getElasticSearchConfig().elasticsearchTemplate(env);
 	}
 	
 	@Bean

--- a/src/test/java/integration/in/ac/bits/protocolanalyzer/mvc/model/ExperimentTest.java
+++ b/src/test/java/integration/in/ac/bits/protocolanalyzer/mvc/model/ExperimentTest.java
@@ -10,10 +10,8 @@ import static org.hamcrest.MatcherAssert.assertThat;
 
 import org.json.JSONObject;
 
-import org.junit.Before;
 import org.junit.Test;
 import org.junit.runner.RunWith;
-import org.mockito.MockitoAnnotations;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Qualifier;
 import org.springframework.test.context.ContextConfiguration;
@@ -33,11 +31,6 @@ public class ExperimentTest{
 	@Autowired
 	@Qualifier("Experiment")
 	private Experiment experiment;
-	
-    @Before
-    public void setUp() {
-        MockitoAnnotations.initMocks(this);
-    }
 	
 	@Test
 	public void autowiringTest() {

